mlrun/runtimes/nuclio/api_gateway.py

```diff
@@ -389,10 +392,10 @@ def invoke(
         ):
             # inject access key from env
             auth = NuclioAuthInfo().from_envvar().to_requests_auth()
-
+        url = urljoin(self.invoke_url, path or "")
```

Comment:
This is already done in `self.invoke_url`

Reply:
the `ApplicationRuntime`'s `invoke` function accepts a required `path` argument.
This is due to the way `invoke` is defined the parent class `RemoteRuntime`.

So, we could potentially have a sub-path to invoke with, regardless of the path in the API Gateway's spec.

Reply:
@TomerShor gotcha, however `self.invoke_url` already does `urljoin(host, self.spec.path)`. Does it mean that we will do urljoin twice?

Reply:
Yes, in the following example case:
- Host: "my.domain.com"
- API Gateway path: "my-path"
- Call `invoke("/some-sub-path")`

The result URL to be called should be `"my.domain.com/my-path/some-sub-path"`